src/components/Footer.tsx

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useCallback, useContext, useMemo } from 'react';
+import classNames from 'classnames';
+import { TodoContext } from '../contexts/TodoContext';
+import { deleteTodo } from '../api/todos';
+import { Filtering } from '../types/Filtering';
+import { Error } from '../types/Error';
+import { getItemsLeft } from '../utils/getItemsLeft';
+
+export const Footer = () => {
+  const {
+    todos,
+    setTodos,
+    filtering,
+    setFiltering,
+    setLoadingTodo,
+    setErrorMessage,
+  } = useContext(TodoContext);
+
+  const itemsLeft = getItemsLeft(todos);
+
+  const completedTodosIds = useMemo(
+    () => todos.filter(todo => todo.completed), [todos],
+  );
+
+  const handleFinally = () => {
+    setLoadingTodo(null);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
```

Comment:
you don't need `setTimeout` here bacause you have already implemented that in `Errors` component